### ISSUE TITLE
Make param type of `BlobApi.getUrl()` less strict?

### ISSUE THREAD

**Issue by achou11** [digidem/comapeo-core#483]
## Description

Currently doing some frontend work that uses the BlobApi to get the http url for a specified observation attachment. I'm having a difficult time appeasing TypeScript due to the `getUrl(blobId)` expecting a `BlobId` type, which is more precise (uses a discriminate union) and notably differs from an attachment type as described by `Observation['attachments'][number]` in mapeo schema.

The `getUrl(blobId)` works great if passing a value created by `BlobApi.create()` (as demonstrated in our tests), but given the type challenges when using the context of observation attachments, I'm wondering if we could update the param type to avoid convoluted typescript-specific code in the consuming application. Perhaps there's something I'm missing in terms of usage that is giving me problems with Typescript but my initial thought is that it would probably be easier to just update the param type (and maybe introduce runtime validation if really necessary)

**Comment by gmaclennan** [comment]
What are the differences vs the attachment type? I thought it was only that the blobId has a variant property, which is necessary.

**Comment by achou11** [comment]
> What are the differences vs the attachment type? I thought it was only that the blobId has a variant property, which is necessary.

Yeah that's the primary difference in terms of defined fields, but I _think_ there's something about how `BlobId` is defined that may be giving me some trouble? Given the following example:

```ts
export function someFunction(
  attachment: Observation['attachments'][number],
  variant: BlobVariant<
    // TODO: Ideally derive this from `attachment` param or something that looks simpler...
    Exclude<Observation['attachments'][number]['type'], 'UNRECOGNIZED'>
  >,
) {
  return project.$blobs.getUrl({
      driveId: attachment.driveDiscoveryId,
      name: attachment.name,
      type: attachment.type,
      variant,
  })
}
```

I'm getting the following error:

```
Argument of type '{ driveId: string; name: string; type: "photo" | "video" | "audio"; variant: BlobVariant<"photo" | "video" | "audio">; }' is not assignable to parameter of type 'BlobId'.
  Types of property 'type' are incompatible.
    Type '"photo" | "video" | "audio"' is not assignable to type '"photo"'.
      Type '"video"' is not assignable to type '"photo"'.ts(2345)
```

Wish I could provide a TS playground link but importing the alpha version of mapeo schema doesn't seem feasible

**Comment by gmaclennan** [comment]
Yeah it's because each type has different variants available, so the type is strict to stop you requesting a variant that does not exist for the given type.

**Comment by achou11** [comment]
> Yeah it's because each type has different variants available, so the type is strict to stop you requesting a variant that does not exist for the given type.

Yeah he reasoning makes sense - is the only solution to do some (somewhat tedious) runtime checks before passing the arguments?

**Comment by gmaclennan** [comment]
Will address by including urls to all available variants in the attachments array for each observation.
